whd/dispatcher.py

```diff
@@ -107,6 +110,14 @@ def dispatch_pullrequest_event(self, pr_event):
                 concourse_api=concourse_api,
                 event=pr_event,
             ))
+
+            if len(resources) == 0:
+                continue
+
+            if pr_event.action() is PullRequestAction.OPENED:
+                required_labels = {resource.source['label'] for resource in resources}
```

Comment:
handle case of absent label attr (hint: use dict.get('label', None) + if not None)

Reply:
also: please mv to separate method to ensure consistent level of abstraction